vector/src/test/java/im/vector/app/features/settings/font/FontScaleSettingViewModelTest.kt

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.settings.font
+
+import com.airbnb.mvrx.test.MvRxTestRule
+import im.vector.app.test.fakes.FakeConfiguration
+import im.vector.app.test.fakes.FakeFontScalePreferences
+import im.vector.app.test.test
+import kotlinx.coroutines.test.runTest
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class FontScaleSettingViewModelTest {
+
+    @get:Rule
+    val mvrxTestRule = MvRxTestRule()
+
+    private val fakeConfiguration = FakeConfiguration()
+    private val fakeFontScalePreferences = FakeFontScalePreferences()
+
+    private var initialState = FontScaleSettingViewState()
+    private lateinit var viewModel: FontScaleSettingViewModel
+
+    @Before
+    fun setUp() {
+        viewModelWith(initialState)
+    }
+
+    private fun viewModelWith(state: FontScaleSettingViewState) {
+        FontScaleSettingViewModel(
+                state,
+                fakeConfiguration.instance,
+                fakeFontScalePreferences
+        ).also {
+            viewModel = it
+            initialState = state
+        }
+    }
+
+    @Test
+    fun `when handling FontScaleChangedAction, then changes state and emits RestartActivity event`() = runTest {
+        val scaleOptions = fakeFontScalePreferences.getAvailableScales()
```

Comment:
ideally our tests would only make real calls to the method currently under test, which for this test case is `ViewModel.handle`

by calling methods on other classes we're coupling our tests to them, if `FontScalePreferences.getAvailableScales` were to change, this test would also need to be updated

something else to bear in mind is which information is needed by the test, for this test case we want to know if given a list of options, when selecting one, is the state updated, restart emitted and the selection persisted, which means we can simplify to...

```kotlin
// A reusable fixture for the given data
private val A_SELECTION = aFontScaleValue(index = 1)

// the ViewModel tests don't need to know about the real font scale options
private val A_SCALE_OPTIONS_WITH_SELECTION = listOf(
        aFontScaleValue(index = 0),
        A_SELECTION,
)

@Test
fun `given scale options containing selection, when handling FontScaleChangedAction with selection, then persists selection, updates selected index and emits RestartActivity`() {
    // we setup the fake with our static test data related to the test name
    fakeFontScalePreferences.givenAvailableScaleOptions(A_SCALE_OPTIONS_WITH_SELECTION)
    viewModelWith(initialState)
    val test = viewModel.test()

    viewModel.handle(FontScaleSettingAction.FontScaleChangedAction(A_SELECTION))

    test
            .assertStatesChanges(
                    // the ViewModel mutates the initial state in its `init`
                    initialState.copy(availableScaleOptions = A_SCALE_OPTIONS_WITH_SELECTION),
                    { copy(persistedSettingIndex = A_SELECTION.index) }
            )
            .assertEvents(FontScaleSettingViewEvents.RestartActivity)
            .finish()
    fakeFontScalePreferences.verifyAppScaleFontValue(A_SELECTION)
}

// our tests only make use of the index 
fun aFontScaleValue(index: Int) = FontScaleValue(index, "foo", -1f, 0)
```